main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala

```diff
@@ -91,6 +91,13 @@ object Simplifier {
       val t = visitType(tpe)
       SimplifiedAst.Expr.Apply(e, es, t, simplifyEffect(eff), loc)
 
+    case MonoAst.Expr.ApplyLocalDef(sym, exps, itpe, tpe, eff, loc) =>
+      val ft = visitType(itpe)
+      val e = SimplifiedAst.Expr.Var(sym, ft, loc)
+      val es = exps.map(visitExp)
+      val t = visitType(tpe)
```

Comment:
Just to repeat what we discussed - this `t` should be curried if we go back into a general apply